modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/converter/visitor/ConditionExpressionVisitor.java

```diff
@@ -103,8 +104,14 @@ public Object visit(FunctionCallExpression functionCallExpression) {
 			SPIDDMFormRuleCondition.Operand operand = doVisit(
 				parameterExpressions.get(1));
 
+			String operandType = operand.getType();
+
+			if (StringUtil.equals(operand.getType(), "string")) {
+				operandType = "option";
+			}
+
 			return new SPIDDMFormRuleCondition.Operand(
-				"option", operand.getValue());
+				operandType, operand.getValue());
```

Comment:
Curiosity: Is there other operand type besides "option" ?
I'm curious why we need to edit the way it was in the code ("option" hard-coded).

Reply:
These types are defined in [this](https://github.com/liferay/liferay-portal/blob/d60700a887fb5d7c57e142b4464eda04a4bef526/modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rules/editor/config.es.js#L14-L20) FE code (hard-coded)